ManatiCR/vitaminau
# VIT-23:  Servicios laboratorio form

# Task [VIT-23](https://manati.atlassian.net/browse/VIT-23?atlOrigin=eyJpIjoiOTI1NWY5NjRhZGIwNDhlZDkyNGY4ODFmMjNkMzVjOGEiLCJwIjoiaiJ9)
Creation of the "Servicios de laboratorio" form.

#Installation
- Run the following command
`ahoy drush cr && ahoy drush cim -y`

# Test
- Go to the following url: 
  - http://vitaminau.docker/user
- Login as admin user:
  - User: admin
  - Pass: admin
- Go to the following url: 
  - http://vitaminau.docker/form/laboratory-services
- Once on that page, you should be able to see a form that contains all the fields specified in the Jira:
  - [VIT-23](https://manati.atlassian.net/browse/VIT-23?atlOrigin=eyJpIjoiOTI1NWY5NjRhZGIwNDhlZDkyNGY4ODFmMjNkMzVjOGEiLCJwIjoiaiJ9)
- Fill all the fields of the form and press the save buttom
- Once you press it your information should be saved and the form should redirect you to a thank you page
- Save your draft with the "Save draft" button and you should see the "El borrador se ha guardado con éxito. Los borradores se almacenarán por un máximo de 2 semanas desde el momento de su creación. Los borradores eliminados no podrán ser restaurados" message and then log back in to another browser or incognito, a "Este formulario ha sido precargado con la información almacenada en el borrador." the message should appear.
- To test the required fields, leave a blank, which has a red "*" and the same form will tell you that it has not been completed yet.

# Acceptance criteria
- Only authenticated users can access the form.
- Form can be saved as draft.
- When the required fields are empty, you should see a message, indicating that it must be filled
- I can complete the form and send it if I complete it correctly.
- Verify that the beneficiary section is the new one.
- Check that all fields with long texts have a character counter.
- The following fields are not required
  - Provincia -> Cantones -> Otras señas
  - Requerimientos
  - Restricciones
  - Detalle
  - ¿Algo que agregar?
  - PDF relevante
  - ¿Cómo calificaría este formulario?
  - ¿Qué mejoraría?
